api/xds_protocol.rst

```diff
@@ -36,43 +36,180 @@ occurs.
-
-Type URLs
-^^^^^^^^^
-
-Each xDS API is concerned with resources of a given type. There is a 1:1
-correspondence between an xDS API and a resource type. That is:
-
--  LDS: :ref:`envoy.api.v2.Listener <envoy_api_msg_Listener>`
--  RDS: :ref:`envoy.api.v2.RouteConfiguration <envoy_api_msg_RouteConfiguration>`
--  VHDS: :ref:`envoy.api.v2.Vhds <envoy_api_msg_RouteConfiguration>`
--  CDS: :ref:`envoy.api.v2.Cluster <envoy_api_msg_Cluster>`
--  EDS: :ref:`envoy.api.v2.ClusterLoadAssignment <envoy_api_msg_ClusterLoadAssignment>`
--  SDS: :ref:`envoy.api.v2.Auth.Secret <envoy_api_msg_Auth.Secret>`
--  RTDS: :ref:`envoy.service.discovery.v2.Runtime <envoy_api_msg_service.discovery.v2.Runtime>`
-
-The concept of `type URLs <https://developers.google.com/protocol-buffers/docs/proto3#any>`_ appears below, and takes the form
-`type.googleapis.com/<resource type>`, e.g.
-`type.googleapis.com/envoy.api.v2.Cluster` for CDS. In various
-requests from Envoy and responses by the management server, the resource
-type URL is stated.
+Resource Types
+~~~~~~~~~~~~~~
+
+Every configuration resource in the xDS API has a type associated with it. The following types are
+supported:
+
+-  :ref:`envoy.api.v2.Listener <envoy_api_msg_Listener>`
+-  :ref:`envoy.api.v2.RouteConfiguration <envoy_api_msg_RouteConfiguration>`
+-  :ref:`envoy.api.v2.ScopedRouteConfiguration <envoy_api_msg_ScopedRouteConfiguration>`
+-  :ref:`envoy.api.v2.route.VirtualHost <envoy_api_msg_route.VirtualHost>`
+-  :ref:`envoy.api.v2.Cluster <envoy_api_msg_Cluster>`
+-  :ref:`envoy.api.v2.ClusterLoadAssignment <envoy_api_msg_ClusterLoadAssignment>`
+-  :ref:`envoy.api.v2.Auth.Secret <envoy_api_msg_Auth.Secret>`
+-  :ref:`envoy.service.discovery.v2.Runtime <envoy_api_msg_service.discovery.v2.Runtime>`
+
+The concept of `type URLs <https://developers.google.com/protocol-buffers/docs/proto3#any>`_
+appears below, and takes the form `type.googleapis.com/<resource type>` -- e.g.,
+`type.googleapis.com/envoy.api.v2.Cluster` for a `Cluster` resource. In various requests from
+Envoy and responses by the management server, the resource type URL is stated.
+
+API flow
+~~~~~~~~
+
+For typical HTTP routing scenarios, the core resource types for the client's configuration are
+`Listener`, `RouteConfiguration`, `Cluster`, and `ClusterLoadAssignment`. Each `Listener` resource
+may point to a `RouteConfiguration` resource, which may point to one or more `Cluster` resources,
+and each Cluster` resource may point to a `ClusterLoadAssignment` resource.
+
+Envoy fetches all `Listener` and `Cluster` resources at startup. It then fetches whatever
+`RouteConfiguration` and `ClusterLoadAssignment` resources that are required by the `Listener` and
+`Cluster` resources. In effect, every `Listener` or `Cluster` resource is a root to part of Envoy's
+configuration tree.
+
+A non-proxy client such as gRPC will start by fetching only the specific `Listener` resources
```

Comment:
nit: s/will start/might start (to account for the multiple listener Envoy Mobile, etc. case)

Reply:
Done.